src/ray/protobuf/gcs.proto

```diff
@@ -200,6 +201,77 @@ message ProfileTableData {
   repeated ProfileEvent profile_events = 4;
 }
 
+message ProfileEventEntry {
+  // The start timestamp of the event time.
+  int64 start_time = 1;
+  // The end timestamp of the event. Empty if it's a point event, e.g. TaskStatus
+  // change.
+  int64 end_time = 2;
+  // Additional data associated with the event. This data must be serialized
+  // using JSON.
+  string extra_data = 3;
+  // Customized event name if not a TaskStatus change point event.
+  string event_name = 4;
+}
+
+message ProfileEvents {
+  // The type of the component that generated the event, e.g., worker or
+  // object_manager, or node_manager.
+  string component_type = 1;
+  // An identifier for the component that generated the event.
+  bytes component_id = 2;
+  // Node IP address.
+  string node_ip_address = 3;
+  // Events.
+  repeated ProfileEventEntry events = 4;
```

Comment:
Do we bound the number of profile events per task?

Reply:
We don't - but we could only enforce the max number of profile events sent in a protobuf within a report interval.